test/components/plots/stackedPlotTests.ts

```diff
@@ -19,145 +19,145 @@ describe("Plots", () => {
-      var ds2PlotMetadata = <Plottable.Plots.StackedPlotMetadata>(<any> stackedPlot)._key2PlotDatasetKey.get("d2").plotMetadata;
+      stackedPlot.addDataset(new Plottable.Dataset(data0));
+      stackedPlot.addDataset(new Plottable.Dataset(data1));
 
-      assert.isTrue(isNaN(ds1PlotMetadata.offsets.get("1")), "stacking is initially incorrect");
+      var keys = (<any> stackedPlot)._key2PlotDatasetKey.keys(); // HACKHACK: Dataset keys are being removed, so these are internal keys
+      var ds0PlotMetadata = <Plottable.Plots.StackedPlotMetadata>(<any> stackedPlot)._key2PlotDatasetKey.get(keys[0]).plotMetadata;
+      var ds1PlotMetadata = <Plottable.Plots.StackedPlotMetadata>(<any> stackedPlot)._key2PlotDatasetKey.get(keys[1]).plotMetadata;
+
+      assert.isTrue(isNaN(ds0PlotMetadata.offsets.get("1")), "stacking is initially incorrect");
 
       stackedPlot.project("x", "a");
       stackedPlot.project("y", "b");
 
-      assert.strictEqual(ds2PlotMetadata.offsets.get("1"), 2, "stacking was done correctly");
+      assert.strictEqual(ds1PlotMetadata.offsets.get("1"), 2, "stacking was done correctly");
     });
 
     it("strings are coerced to numbers for stacking", () => {
-      var data1 = [
+      var data0 = [
         { x: 1, y: "-2" }
       ];
-      var data2 = [
+      var data1 = [
         { x: 1, y: "3" }
       ];
-      var data3 = [
+      var data2 = [
         { x: 1, y: "-1" }
       ];
-      var data4 = [
+      var data3 = [
         { x: 1, y: "5" }
       ];
-      var data5 = [
+      var data4 = [
         { x: 1, y: "1" }
       ];
-      var data6 = [
+      var data5 = [
         { x: 1, y: "-1" }
       ];
 
-      stackedPlot.addDataset("d1", data1);
-      stackedPlot.addDataset("d2", data2);
-      stackedPlot.addDataset("d3", data3);
-      stackedPlot.addDataset("d4", data4);
-      stackedPlot.addDataset("d5", data5);
-      stackedPlot.addDataset("d6", data6);
+      stackedPlot.addDataset(new Plottable.Dataset(data0));
+      stackedPlot.addDataset(new Plottable.Dataset(data1));
+      stackedPlot.addDataset(new Plottable.Dataset(data2));
+      stackedPlot.addDataset(new Plottable.Dataset(data3));
+      stackedPlot.addDataset(new Plottable.Dataset(data4));
+      stackedPlot.addDataset(new Plottable.Dataset(data5));
 
-      var ds3PlotMetadata = <Plottable.Plots.StackedPlotMetadata>(<any> stackedPlot)._key2PlotDatasetKey.get("d3").plotMetadata;
-      var ds4PlotMetadata = <Plottable.Plots.StackedPlotMetadata>(<any> stackedPlot)._key2PlotDatasetKey.get("d4").plotMetadata;
-      var ds5PlotMetadata = <Plottable.Plots.StackedPlotMetadata>(<any> stackedPlot)._key2PlotDatasetKey.get("d5").plotMetadata;
-      var ds6PlotMetadata = <Plottable.Plots.StackedPlotMetadata>(<any> stackedPlot)._key2PlotDatasetKey.get("d6").plotMetadata;
+      var keys = (<any> stackedPlot)._key2PlotDatasetKey.keys(); // HACKHACK: Dataset keys are being removed, so these are internal keys
```

Comment:
This is mostly a plotMetadata issue but we should probably ticket this for reference